src/parser.rs

```diff
@@ -465,6 +465,33 @@ impl Parser {
         }
     }
 
+    // Hacked version of parse_date_time_field to allow directly passing strs.
+    pub fn parse_date_time_field_given_str(
```

Comment:
I originally tried using this as an internal version for `parse_date_time_field` as well, but hit a problem.

When someone tries to extract a date/time field that **isn't** a SQL keyword, we want to show the mistaken keyword back to them like:
```
    let res = parse_sql_statements("SELECT EXTRACT(MILLISECOND FROM d)");
    assert_eq!(
        ParserError::ParserError("Expected date/time field, found: MILLISECOND".to_string()),
        res.unwrap_err()
    );
```

If everything went through `parse_date_time_field_given_str`, `Word`s with non-keywords would not have a `&str` value, resulting in a printed error like:
```
Expected date/time field, found: 
```